### DISCUSSION TITLE
Guidance on Contributions & Project

### DISCUSSION THREAD

**Discussion by ryanmerolle** [napalm-automation/napalm-logs#383] [General]
**Housekeeping items:**
1. Are you ok with me including a devcontainer in the project?
2. Would you be against me using poetry(w/venv) and replace requirements files?
3. Would you be ok with me adding a workflow to publish containers to ghcr also?
4. Would you be ok with me triggering weekly builds and publishes of the latest tag to docker hub & ghcr? (Relates to pin requirements and poetry)
5. It feels like there are a lot of open issues that looked to have been completed.  Is that something I should just comment on to see if we can close them as is, or if they are even worth me looking at?

**App questions:**
1. Is there anyway to send UNKNOWN and RAW logs to separate log files so I can look to build YAML Profiles for anything I find while testing?  If not maybe I can create a publishers that just writes to a file.
2. As a noob to openconfig/yang, where do I look to easily see how you determined "BGP_NEIGHBOR_STATE_CHANGED" as an error type?  It totally makes sense as an error, but I assume you found it in the model somewhere?
3. Anyone use this with elk?  Do you just send to syslog, http, via Kafka?

As always @mirceaulinic thanks for all of this work!

**Comment by mirceaulinic** [comment]
Hi @ryanmerolle, thanks for chiming in. I'll reply, in-line, below:

> Are you ok with me including a devcontainer in the project?

Yes.

> Would you be against me using poetry(w/venv) and replace requirements files?

While in principle I agree, I'll need to chat with the rest of the NAPALM maintainers, as we typically keep the projects aligned.

> Would you be ok with me adding a workflow to publish containers to ghcr also?

For sure.

> Would you be ok with me triggering weekly builds and publishes of the latest tag to docker hub & ghcr? (Relates to pin requirements and poetry)

Go for it.

> It feels like there are a lot of open issues that looked to have been completed. Is that something I should just comment on to see if we can close them as is, or if they are even worth me looking at?

Please do, any help is very appreciated. :)

> Is there anyway to send UNKNOWN and RAW logs to separate log files so I can look to build YAML Profiles for anything I find while testing? If not maybe I can create a publishers that just writes to a file.

Currently not to a file, but you can probably use the [CLI](https://napalm-logs.readthedocs.io/en/latest/publisher/cli.html) Publisher in conjunction with the [YAML](https://napalm-logs.readthedocs.io/en/latest/serializer/yaml.html) Serializer and pipe that output. But a new Publisher that writes to a file (or more, per message type) would be a nice addition too! :)

Configuration example (for CLI + YAML):

```yaml
publisher:
  - cli:
      send_raw: true
      serializer: yaml
```

> As a noob to openconfig/yang, where do I look to easily see how you determined "BGP_NEIGHBOR_STATE_CHANGED" as an error type? It totally makes sense as an error, but I assume you found it in the model somewhere?

We typically try to find the most appropriate YANG model using https://www.yangcatalog.org/, as it seems sensible. If in doubt for one or more specific use-cases, ask away.

> Anyone use this with elk? Do you just send to syslog, http, via Kafka?

I did via Kafka, and currently syslog. But the possibilities are virtually endless with the existing Publisher modules or others that we can proably craft.

Let me know if you have any further questions. 😃

**Reply by ryanmerolle** [reply]
I am familiar with the options for cli publisher.  It just sort of got lost in my debugs I run at the same time.  I will look into the file publisher.
